qemu/tests/numa_cpu.py

```diff
@@ -0,0 +1,149 @@
+import logging
+
+from virttest import error_context
+from virttest.utils_misc import NumaInfo
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Assign cpu to numa node with "-numa cpu", check the numa info in monitor
+    and guest os match with the qemu cli
+    """
+
+    def convert_cpu_topology_to_index(socket, die, core, thread):
```

Comment:
Maybe we could maintain a list of all cpus,  every item is a list [socketid, dieid, coreid, threadid], like [[0,0,0,0], [0,0,0,1]...]. The index is the id of every cpu. Not sure if it meets all the requirement, just a thought.

Reply:
> Maybe we could maintain a list of all cpus, every item is a list [socketid, dieid, coreid, threadid], like [[0,0,0,0], [0,0,0,1]...]. The index is the id of every cpu. Not sure if it meets all the requirement, just a thought.

From the monitor, `info numa` final output is the cpu index, and cpu info from guest os use index too. 
cpu index and topology are one-to-one,  so i use cpu index as the target to compare instead of using the cpu topology.  By this was, we can use the API from framework without transfer.

Reply:
Yeah, I understand. I meant, with the list(maybe tuple is better ), we can find the index quickly and don't need to calculate index for every cpu.
```
>>> b=([0,0,0],[0,0,1],[0,1,0],[0,1,1])
>>> b.index([0,0,1])
1
```

Reply:
The index is only the index in the list, does it related with the actual cpu index in guest os ?

Reply:
Generate the list b is another work, track this issue with the comment below.